[google_fonts] Extract the config class to its own file in prevision of a GoogleFontsLite version (#11602)
flutter/packages: packages/google_fonts/CHANGELOG.md

## 8.2.0

> - Extract the class `Config` to its own file and rename it `GoogleFontsConfig`. The `Config` name will be deprecated in a future release.

Reviewer: It is deprecated though in this change, isn't it?

Reviewer: ```suggestion
- Extract the class `Config` to its own file and rename it `GoogleFontsConfig`. The `Config` class is now deprecated.
```
